### ISSUE TITLE
Can't build cache with new theme without creating cache dir

### ISSUE THREAD

**Issue by justmendes** [sharkdp/bat#576]
In a fresh install of both macOS (10.14.5) and bat, I created a new theme and followed the steps on the README to install it.

However running `bat cache --build` returned the following error:

```shell
$ bat cache --build

No syntaxes were found in '/Users/frm/.config/bat/syntaxes', using the default set.
[bat error]: Could not save theme set to /Users/frm/.cache/bat/themes.bin
```

I was able to solve it by running `mkdir -p $(bat --cache-dir)`.

You can find all my bat configs in [my dotfiles](github.com/justmendes/dotfiles), if relevant.

The bat install was done by an automated script. [Relevant lines](https://github.com/justmendes/dotfiles/blob/master/install#L202-L207)

**Comment by sharkdp** [comment]
Thank you for reporting this.

In principle, `bat` should create the cache dir by itself:
https://github.com/sharkdp/bat/blob/50aabf331dc29201560fecd1b478110964736db5/src/assets.rs#L120

However, this will fail if the parent directory (in your case: `/Users/frm/.cache`) does not exist. Is it possible that you did not have a `/Users/frm/.cache` directory before calling `bat cache --build`?

If this is the case, I think this could be easily fixed by using [`fs::create_dir_all`](https://doc.rust-lang.org/std/fs/fn.create_dir_all.html) instead of `fs::create_dir`.

**Comment by justmendes** [comment]
That's precisely it, forgot to mention it. I noticed I didn't have a `~/.cache` dir, which led me down that path. 👍 If you want, I can take this issue.

**Comment by sharkdp** [comment]
Gladly!

**Comment by sharkdp** [comment]
Fixed in [bat v0.12](https://github.com/sharkdp/bat/releases/tag/v0.12.0).
